src/net_processing.cpp

```diff
@@ -1063,12 +1067,63 @@ bool static AlreadyHave(const CInv& inv) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
 static void RelayTransaction(const CTransaction& tx, CConnman* connman)
 {
     CInv inv(MSG_TX, tx.GetHash());
-    connman->ForEachNode([&inv](CNode* pnode)
-    {
+    connman->ForEachNode([&inv](CNode* pnode) {
         pnode->PushInventory(inv);
+        pnode->m_cache_dandelion.erase(inv.hash);
     });
 }
 
+static void FluffTransaction(CTxMemPool& tx_pool, const CNode* from, const CTransactionRef& ptx, CConnman* connman)
+{
+    LogPrint(BCLog::DANDELION, "fluff\n");
+    CValidationState state;
+    if (!AcceptToMemoryPool(tx_pool, state, ptx, /* pfMissingInputs */ nullptr, /* plTxnReplaced */ nullptr, /* bypass_limits */ false, /* nAbsurdFee */ 0, /* test_accept */ false)) {
+        // Should never happen, since we checked for validity
+        LogPrint(BCLog::DANDELION, "Dandelion transaction fluff failed due to %s\n", FormatStateMessage(state));
+        return;
+    }
+    tx_pool.check(pcoinsTip.get());
+    RelayTransaction(*ptx, connman);
+    LogPrint(BCLog::MEMPOOL, "AcceptToMemoryPool: peer=%d: accepted %s (poolsz %u txn, %u kB)\n",
+        from->GetId(),
+        ptx->GetHash().ToString(),
+        tx_pool.size(), tx_pool.DynamicMemoryUsage() / 1000);
+}
+
+static void RelayDandelionTransaction(CTxMemPool& tx_pool, CNode* node_from, const CTransactionRef& ptx, const CNetMsgMaker& msg_maker, CConnman* connman)
+{
+    bool fluff = GetRand(100 * 100) / 100. >= CNode::m_dandelion_stem_pct_threshold;
+    if (fluff) return FluffTransaction(tx_pool, node_from, ptx, connman);
+
+    CNode* dest = node_from->GetDandelionDestination(connman);
+    if (!dest) {
+        // Just fall back to full fluff at this point
+        return FluffTransaction(tx_pool, node_from, ptx, connman);
+    }
+
+    // Dandelion relay (at least one hop)
+    const int64_t now_micros = GetTimeMicros();
+    const int64_t expiry = now_micros + EXPIRE_DANDELION_CACHE * 1000 * 1000 + PoissonNextSend(now_micros, EXPIRE_DANDELION_CACHE_AVG_ADD);
+    node_from->m_cache_dandelion.emplace(std::make_pair(ptx->GetWitnessHash(), std::make_pair(ptx, expiry)));
+    node_from->m_cache_dandelion_size += ptx->GetTotalSize();
+    node_from->m_cache_expiry.push(expiry);
+
+    static constexpr int DANDELION_SEND_FLAGS{0}; // Always send as witness tx
+
+    // Check if the next peer would fluff (assume same probability)
+    bool force_fluff = GetRand(100 * 100) / 100. >= CNode::m_dandelion_stem_pct_threshold;
```

Comment:
This doubles the odds of the next node fluffing -- 10% chance that we thought they should fluff, but if we don't think they should fluff, they'll still make their own decision, giving another 9% (=90%*10%) chance that they'll think they should fluff for a total of 19%.

(necroreviewing!)